src/System.ComponentModel.TypeConverter/src/System/Drawing/RectangleConverter.cs

```diff
@@ -0,0 +1,171 @@
+                {
+                    // Parse 4 integer values.
+                    //
+                    if (culture == null)
+                    {
+                        culture = CultureInfo.CurrentCulture;
+                    }
+                    char sep = culture.TextInfo.ListSeparator[0];
+                    string[] tokens = text.Split(new char[] { sep });
+                    int[] values = new int[tokens.Length];
+                    TypeConverter intConverter = TypeDescriptor.GetConverter(typeof(int));
+                    for (int i = 0; i < values.Length; i++)
+                    {
+                        // Note: ConvertFromString will raise exception if value cannot be converted.
+                        values[i] = (int)intConverter.ConvertFromString(context, culture, tokens[i]);
+                    }
+
+                    if (values.Length == 4)
+                    {
+                        return new Rectangle(values[0], values[1], values[2], values[3]);
+                    }
+                    else
+                    {
+                        throw new ArgumentException(SR.Format(SR.TextParseFailedFormat,
+                                                                  "text",
+                                                                  text,
+                                                                  "x, y, width, height"));
+                    }
+                }
+            }
+
+            return base.ConvertFrom(context, culture, value);
+        }
+
+        public override object ConvertTo(ITypeDescriptorContext context, CultureInfo culture, object value, Type destinationType)
+        {
+            if (destinationType == null)
+            {
+                throw new ArgumentNullException(nameof(destinationType));
+            }
+
+            if (value is Rectangle)
+            {
+                if (destinationType == typeof(string))
+                {
+                    Rectangle rect = (Rectangle)value;
+
+                    if (culture == null)
+                    {
+                        culture = CultureInfo.CurrentCulture;
+                    }
+                    string sep = culture.TextInfo.ListSeparator + " ";
+                    TypeConverter intConverter = TypeDescriptor.GetConverter(typeof(int));
+                    string[] args = new string[4];
+                    int nArg = 0;
+
+                    // Note: ConvertToString will raise exception if value cannot be converted.
+                    args[nArg++] = intConverter.ConvertToString(context, culture, rect.X);
+                    args[nArg++] = intConverter.ConvertToString(context, culture, rect.Y);
+                    args[nArg++] = intConverter.ConvertToString(context, culture, rect.Width);
+                    args[nArg++] = intConverter.ConvertToString(context, culture, rect.Height);
+
+                    return string.Join(sep, args);
+                }
+#if FEATURE_INSTANCEDESCRIPTOR
```

Comment:
This was already in the other converters around the code that deals with InstanceDescriptor. We can remove these ifdefs once InstanceDescriptor is ported.